test/integration/test_kongplugin.go

```diff
@@ -0,0 +1,202 @@
+			Namespace: ns.Name,
+		},
+		PluginName: "proxy-cache",
+		Config: apiextensionsv1.JSON{
+			Raw: []byte(`{"response_code": [200], "request_method": ["GET", "HEAD"], "content_type": ["text/plain; charset=utf-8"], "cache_ttl": 300, "strategy": "memory"}`),
+		},
+	}
+	err = GetClients().MgrClient.Create(GetCtx(), proxyCachekongPlugin)
+	require.NoError(t, err)
+	cleaner.Add(proxyCachekongPlugin)
+
+	t.Log("deploying a KongService resource")
+	ksName := "ks-" + testID
+	kongService := &configurationv1alpha1.KongService{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ksName,
+			Namespace: ns.Name,
+			Annotations: map[string]string{
+				"konghq.com/plugins": rateLimitingkongPluginName,
+			},
+		},
+		Spec: configurationv1alpha1.KongServiceSpec{
+			KongServiceAPISpec: configurationv1alpha1.KongServiceAPISpec{
+				Name: lo.ToPtr(ksName),
+				URL:  lo.ToPtr("http://example.com"),
+			},
+		},
+	}
+	err = GetClients().MgrClient.Create(GetCtx(), kongService)
+	require.NoError(t, err)
+	cleaner.Add(kongService)
+
+	t.Log("waiting for the managed KongPluginBinding to be created")
+	managedKPB := &configurationv1alpha1.KongPluginBinding{}
+	require.Eventually(t, func() bool {
+		kongPluginBindingList := &configurationv1alpha1.KongPluginBindingList{}
+		err := GetClients().MgrClient.List(GetCtx(), kongPluginBindingList, client.InNamespace(ns.Name))
+		if err != nil {
+			return false
+		}
+		if len(kongPluginBindingList.Items) != 1 {
+			return false
+		}
+		managedKPB = &kongPluginBindingList.Items[0]
+		return true
+	}, 30*time.Second, time.Second)
+
+	t.Log("delete managed kongPluginBinding, then check it gets recreated")
+	err = GetClients().MgrClient.Delete(GetCtx(), managedKPB)
+	require.NoError(t, err)
+	require.Eventually(t, func() bool {
+		kongPluginBindingList := &configurationv1alpha1.KongPluginBindingList{}
+		err := GetClients().MgrClient.List(GetCtx(), kongPluginBindingList, client.InNamespace(ns.Name))
+		if err != nil {
+			return false
+		}
+		if len(kongPluginBindingList.Items) != 1 {
+			return false
+		}
+		managedKPB = &kongPluginBindingList.Items[0]
+		return true
+	}, 30*time.Second, time.Second)
+
+	t.Log("remove annotation from kongservice")
+	newKongService := kongService.DeepCopy()
+	newKongService.Annotations = nil
+	require.Eventually(t, func() bool {
+		if err := GetClients().MgrClient.Patch(GetCtx(), newKongService, client.MergeFrom(kongService)); err != nil {
+			return false
+		}
+		return true
+	}, 30*time.Second, 5*time.Second)
+
+	t.Log("then check the managed kpb gets deleted")
+	require.Eventually(t, func() bool {
+		kongPluginBindingList := &configurationv1alpha1.KongPluginBindingList{}
+		err := GetClients().MgrClient.List(GetCtx(), kongPluginBindingList, client.InNamespace(ns.Name))
+		if err != nil {
+			return false
+		}
+		if len(kongPluginBindingList.Items) > 0 {
+			return false
+		}
+		return true
+	}, 15*time.Second, time.Second)
+
+	t.Log("deploying an unmanaged KongPluginBinding resource")
```

Comment:
Log the KPB name.

Reply:
6ac46b88b163193a309eaaab096317cb6e743a28